src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/compute/v4_9/CloudStackComputePlugin.java

```diff
@@ -337,33 +329,30 @@ ComputeInstance getComputeInstance(@NotNull GetVirtualMachineResponse.VirtualMac
     }
 
     @VisibleForTesting
-    int getVirtualMachineDiskSize(String virtualMachineId, @NotNull CloudStackUser cloudUser)
-            throws FogbowException {
-
-        GetVolumeRequest request = new GetVolumeRequest.Builder()
-                .virtualMachineId(virtualMachineId)
-                .type(DEFAULT_VOLUME_TYPE_VALUE)
-                .build(this.cloudStackUrl);
-        URIBuilder uriRequest = request.getUriBuilder();
-        String token = cloudUser.getToken();
-        CloudStackUrlUtil.sign(uriRequest, token);
-
-        GetVolumeResponse volumeResponse = null;
+    int getVirtualMachineDiskSize(String virtualMachineId, @NotNull CloudStackUser cloudUser) {
         try {
+            GetVolumeRequest request = new GetVolumeRequest.Builder()
+                    .virtualMachineId(virtualMachineId)
+                    .type(DEFAULT_VOLUME_TYPE_VALUE)
+                    .build(this.cloudStackUrl);
+
+            URIBuilder uriRequest = request.getUriBuilder();
+            String token = cloudUser.getToken();
+            CloudStackUrlUtil.sign(uriRequest, token);
             String jsonResponse = doGet(uriRequest.toString(), cloudUser);
-            volumeResponse = GetVolumeResponse.fromJson(jsonResponse);
-        } catch (HttpResponseException e) {
-            CloudStackHttpToFogbowExceptionMapper.map(e);
-        }
+            GetVolumeResponse response = GetVolumeResponse.fromJson(jsonResponse);
 
-        List<GetVolumeResponse.Volume> volumes = volumeResponse.getVolumes();
-        if (volumes != null && !volumes.isEmpty()) {
-            GetVolumeResponse.Volume firstVolume = volumes.get(0);
-            long sizeInBytes = firstVolume.getSize();
-            return convertBytesToGigabyte(sizeInBytes);
-        } else {
-            throw new InstanceNotFoundException();
+            List<GetVolumeResponse.Volume> volumes = response.getVolumes();
+            if (!volumes.isEmpty()) {
+                GetVolumeResponse.Volume firstVolume = volumes.get(0);
+                long sizeInBytes = firstVolume.getSize();
+                return convertBytesToGigabyte(sizeInBytes);
+            }
+        } catch (Exception e) {
+            LOGGER.debug("Error while trying to get the virtual machine size", e);
```

Comment:
shouldn't it throw an exception?

Reply:
I followed the previous decision. 
I think that the idea was the disk data is not important enough to fail the whole request. 
Do you want to debate more about this?